api/proto/teleport/dbobject/v1/dbobject.proto

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.dbobject.v1;
+
+import "teleport/header/v1/metadata.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/dbobject/v1;dbobjectv1";
+
+// DatabaseObject represents a database object that can be imported into Teleport.
+message DatabaseObject {
+  // The kind of resource represented.
+  string kind = 1;
+  // Differentiates variations of the same kind. All resources should
+  // contain one, even if it is never populated.
+  string sub_kind = 2;
+  // The version of the resource being represented.
+  string version = 3;
+  // Common metadata that all resources share.
+  teleport.header.v1.Metadata metadata = 4;
+  // Specification for the database object.
+  DatabaseObjectSpec spec = 5;
+}
+
+// DatabaseObjectSpec is the spec for the database object.
+message DatabaseObjectSpec {
+  // The protocol used to connect to the database (e.g., postgres).
```

Comment:
Are all fields required? Please document on which field whether it's required or optional.

Reply:
Is this really a relevant information to put here? What makes an object valid tends to be complex, often only easily express as a code. Just take a look on `message RoleConditions` and how it doesn't speak about validation for each field, while in practice the rules are in place, and pretty complex too.

Reply:
> Is this really a relevant information to put here?

Yes, I think so. I'm not asking you to express complex scenarios, although arguably these are even more valuable to document. If it's simple to say whether a field is optional or required, that is helpful information for users and readers of your API. Think of it as part of the contract.

For example, take a look at the last comment line in every field here: https://github.com/gravitational/teleport/blob/30e0e2f1baf83c8ae3f08999be7aa6b27ecbdcc2/api/proto/teleport/devicetrust/v1/device.proto#L41-L43

This is simple to do and clearly documents the API expectations.